File: ruby/raindrops/raindrops.rb
class Raindrops
  
  require 'prime'
  
  def self.convert(number)
    raindrop_speak = { 3 => 'Pling', 5 => 'Plang', 7 => 'Plong' }
    
    raindrop_factors = number.prime_division.map(&:first) & raindrop_speak.keys
    
    return number.to_s if raindrop_factors.empty?
    
    raindrop_factors.inject("") { |s, factor| s << raindrop_speak[factor] }
  end
  
end

module BookKeeping
  VERSION = 3
end

File: go/raindrops/raindrops.go
package raindrops

import "strconv"

// Convert converts a number to a string
func Convert(n int) string {
	s := ""

	if n%3 == 0 {
		s = s + "Pling"
	}

	if n%5 == 0 {
		s = s + "Plang"
	}

	if n%7 == 0 {
		s = s + "Plong"
	}

	if s == "" {
		s = strconv.Itoa(n)
	}

	return s
}
